### ISSUE TITLE
Unable to read number_format with pandas / openpyxl (engine)

### ISSUE THREAD

**Issue by mithunmo** [pandas-dev/pandas#30272]
#### Code Sample

```python
# Your code here
import pandas as pd
data = pd.read_excel('temp.xlsx', engine="openpyxl")
print(data.iloc[0][4].number_format)

######Via openpyxl
from openpyxl import load_workbook
wb = load_workbook('temp.xlsx')
ws = wb.worksheets[0]

col_d = ws.cell(row=2, column=5)
print(col_d.value)
col_nf = col_d.number_format
print(col_nf)

```

#### Problem description

Unable to access number_format in pandas dataframe using engine openpyxl. 
But i am able to acccess the number_format in plain openoyxl via above code.

#### Expected Output
"$"#,##0
#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]
INSTALLED VERSIONS
------------------
commit           : None
python           : 3.6.8.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
machine          : AMD64
processor        : Intel64 Family 6 Model 142 Stepping 9, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.None

pandas           : 0.25.2
numpy            : 1.17.3
pytz             : 2019.2
dateutil         : 2.8.0
pip              : 19.3.1
setuptools       : 41.6.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : 0.9.3
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : 3.0.2
pandas_gbq       : None
pyarrow          : None
pytables         : None
s3fs             : None
scipy            : None
sqlalchemy       : 1.3.10
tables           : None
xarray           : None
xlrd             : 1.2.0
xlwt             : None
xlsxwriter       : None
None

</details>

**Comment by WillAyd** [comment]
`number_format` is a property of certain openpyxl objects. DataFrames store the values of those cells and do not recreate the openpyxl object itself by design

**Comment by legalrealist** [comment]
Security implication: XLSX number format divergence enables parser differential attacks against LLM pipelines

This issue has a security-relevant use case worth raising. Excel's custom number formats can display arbitrary static text regardless of the underlying cell value — a cell containing 146500000 can display $127,400,000 via the format string "$127,400,000". Since pd.read_excel() returns the underlying numeric value and discards format strings, any downstream consumer (including LLM platforms) sees different numbers than what Excel displays.

I built a proof-of-concept demonstrating this end-to-end: a poisoned XLSX where the underlying values are subtly inflated (10-15%) while Excel displays the real figures. Tested on three frontier LLM platforms — all shifted their financial recommendation based on the inflated values. When the same spreadsheet was uploaded as a rendered screenshot, all three read the display values and gave the opposite recommendation. Same file, same prompt, opposite outcome.

Gemini's code interpreter executed pd.read_excel() directly during testing (visible in its code output), confirming this is the extraction path in at least one production LLM platform.

Repo with the proof-of-concept, detection tool, and full writeup: https://github.com/legalrealist/lying-spreadsheets

A minimal fix: an option on read_excel() to surface the number_format metadata alongside the underlying values, so downstream consumers can detect when a format string is a static literal that diverges from the cell value. openpyxl already parses this from xl/styles.xml via cell.number_format — it's just not passed through to pandas.
